orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/tasks/DeleteDeliveryConfigTask.java

```diff
@@ -0,0 +1,66 @@
+package com.netflix.spinnaker.orca.front50.tasks;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.Task;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.front50.Front50Service;
+import com.netflix.spinnaker.orca.front50.model.DeliveryConfig;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import retrofit.client.Response;
+
+import javax.annotation.Nonnull;
+import java.util.Optional;
+
+import static org.springframework.http.HttpStatus.*;
+
+@Component
+public class DeleteDeliveryConfigTask implements Task {
+
+  private Logger log = LoggerFactory.getLogger(getClass());
+
+  private Front50Service front50Service;
+  private ObjectMapper objectMapper;
+  private DeliveryConfigUtils deliveryConfigUtils;
+
+  @Autowired
+  public DeleteDeliveryConfigTask(Front50Service front50Service, ObjectMapper objectMapper) {
+    this.front50Service = front50Service;
+    this.objectMapper = objectMapper;
+    deliveryConfigUtils = new DeliveryConfigUtils(front50Service, objectMapper);
+  }
+
+  @Nonnull
+  @Override
+  public TaskResult execute(@Nonnull Stage stage) {
+    if (!stage.getContext().containsKey("deliveryConfigId")) {
+      throw new IllegalArgumentException("Key 'deliveryConfigId' must be provided.");
+    }
+
+    String configId = stage.getContext().get("deliveryConfigId").toString();
+    Optional<DeliveryConfig> config = deliveryConfigUtils.getDeliveryConfig(configId);
+
+    if (!config.isPresent()) {
+      log.debug("Config {} does not exist, considering deletion successful.", configId);
+      return new TaskResult(ExecutionStatus.SUCCEEDED);
+    }
+
+    try {
+      log.debug("Deleting delivery config: " + objectMapper.writeValueAsString(config.get()));
+    } catch (JsonProcessingException e) {
+      log.debug("Deleting malformed delivery config:" + config.get());
```

Comment:
nit: this can probably not because it's malformed but rather some underlying issue json parser. Not sure if it's worth logging the original exception, we are deleting the config after all..

Reply:
This log statement is here more as a reaction to past issues. People sometimes ask us to revert deletions, and it's handy to be able to find that full object in the logs so that it's easy to revert. 

I'll log the error here just in case.

Reply:
Back-to-back logs also seem odd to me ... the try/catch is super narrowly scoped so I don't know that we need one logging the exception + `config.get()`.

Reply:
@ajordens better if I remove the second logs and just ignore the `JsonProcessingException`?

Reply:
Looking closer, I don't really see the try/catch around an attempted serialization to be all that relevant. 

Seems removable.